client_test.go

```diff
@@ -203,6 +214,54 @@ func Test_Client(t *testing.T) {
 		rivertest.WaitOrTimeout(t, workedChan)
 	})
 
+	t.Run("JobCancel", func(t *testing.T) {
+		t.Parallel()
+
+		client, bundle := setup(t)
+
+		type JobArgs struct {
+			JobArgsReflectKind[JobArgs]
+		}
+
+		AddWorker(client.config.Workers, WorkFunc(func(ctx context.Context, job *Job[JobArgs]) error {
+			return JobCancel(fmt.Errorf("a persisted internal error"))
+		}))
```

Comment:
Should we check the job in the database after this is all done for completeness? i.e. make sure it is cancelled and the error is saved

Reply:
Sure! Added.